server/channels/app/migrations.go

```diff
@@ -570,6 +570,19 @@ func (s *Server) doElasticsearchFixChannelIndex() {
 		mlog.Fatal("failed to start job for fixing Elasticsearch channels index", mlog.Err(appErr))
 		return
 	}
+
+	s.AddLicenseListener(func(oldLicense, newLicense *model.License) {
```

Comment:
This looks very wrong. Doesn't `doElasticsearchFixChannelIndex` return early if the server isn't licensed for ES? So if the server isn't licensed, the listener doesn't get added? And if the listener gets added, the server *is* licensed and the listener will never do anything?